packages/block-library/src/navigation/edit/deleted-navigation-warning.js

```diff
@@ -4,9 +4,16 @@
 import { Warning } from '@wordpress/block-editor';
 import { Button, Notice } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
-import { createInterpolateElement } from '@wordpress/element';
+import { useState, createInterpolateElement } from '@wordpress/element';
 
 function DeletedNavigationWarning( { onCreateNew, isNotice = false } ) {
+	const [ isButtonDisabled, setIsButtonDisabled ] = useState( false );
+
+	const handleButtonClick = () => {
+		setIsButtonDisabled( true );
+		onCreateNew();
+	};
```

Comment:
I think I would personally choose to inline this function and as a nicety also pass the arguments through to `onCreateNew`, as this changes the type/signature of the prop.

It still works fine though, so these changes aren't strictly needed, I leave it as some thoughtful feedback. 😄

Reply:
@talldan  Thank you so much for your feedback! I really appreciate your insight and will definitely keep this in mind for the future.